ui/src/alert-configuration/ConfigureAlerting.test.tsx

```diff
@@ -0,0 +1,1275 @@
+      await userEvent.click(selectors.authSelect.authClick());
+    });
+
+    expect(selectors.enableConfiguration()).not.toBeChecked();
+    expect(selectors.host()).toHaveValue('');
+    expect(selectors.port()).toHaveValue(25);
+    expect(selectors.enableTls()).not.toBeChecked();
+
+    expect(selectors.authSelect.label()).toHaveTextContent('NO AUTHENTICATION');
+    expect(selectors.authSelect.optionNoAuth()).toBeInTheDocument();
+    expect(selectors.authSelect.optionLogin()).toBeInTheDocument();
+    expect(selectors.authSelect.optionCramMd5()).toBeInTheDocument();
+    expect(selectors.authSelect.optionPlain()).toBeInTheDocument();
+
+    expect(selectors.sender()).toHaveValue('');
+    expect(selectors.recipient()).toHaveValue('');
+    expect(selectors.receiveResolved()).not.toBeChecked();
+
+    expect(selectors.sendTestingEmailButton()).toBeEnabled();
+    expect(selectors.cancelButton()).toBeEnabled();
+    expect(selectors.saveButton()).toBeEnabled();
+
+    await act(async () => {
+      await userEvent.click(selectors.authSelect.optionLogin());
+    });
+
+    expect(selectors.authSelect.label()).toHaveTextContent('LOGIN');
+    expect(selectors.username()).toBeInTheDocument();
+    expect(selectors.password()).toBeInTheDocument();
+
+    await act(async () => {
+      await userEvent.click(selectors.authSelect.authClick());
+      await userEvent.click(selectors.authSelect.optionCramMd5());
+    });
+
+    expect(selectors.authSelect.label()).toHaveTextContent('CRAM-MD5');
+    expect(selectors.username()).toBeInTheDocument();
+    expect(selectors.secret()).toBeInTheDocument();
+
+    await act(async () => {
+      await userEvent.click(selectors.authSelect.authClick());
+      await userEvent.click(selectors.authSelect.optionPlain());
+    });
+
+    expect(selectors.authSelect.label()).toHaveTextContent('PLAIN');
+    expect(selectors.identity()).toBeInTheDocument();
+    expect(selectors.username()).toBeInTheDocument();
+    expect(selectors.password()).toBeInTheDocument();
+  });
+
+  it('render form values when the form is defined (NO AUTHENTICATION)', async () => {
+    const configYaml = fs.readFileSync(
+      path.join(__dirname, './test-yaml/simple-noauth.yaml'),
+      'utf8',
+    );
+    server.use(
+      rest.get(
+        `http://localhost/api/kubernetes/api/v1/namespaces/metalk8s-monitoring/configmaps/metalk8s-alertmanager-config`,
+        (req, res, ctx) => {
+          const test = {
+            kind: 'ConfigMap',
+            apiVersion: 'v1',
+            metadata: {
+              name: 'metalk8s-alertmanager-config',
+              namespace: 'metalk8s-monitoring',
+              uid: 'xxx-yyy-zzz',
+              resourceVersion: '0',
+              creationTimestamp: '2023-03-23T16:45:41Z',
+              labels: {},
+              managedFields: [],
+            },
+            data: { 'config.yaml': configYaml },
+          };
+          return res(ctx.json(test));
+        },
+      ),
```

Comment:
nit: maybe we can extract it in a function

Reply:
Yep, totally. Let me do that.